doc/source/debug.rst

```diff
@@ -0,0 +1,191 @@
+show up in this section.
+
+**Real world scenario:** In our case, one of the lines in the events page
+displays an error::
+
+      $ kubectl --namespace=jhub describe pod jupyter-choldgraf
+      ...
+      2m            52s             4       kubelet, gke-jhubtest-default-pool-52c36683-jv6r        spec.containers{notebook}       Warning         Failed           Failed to pull image "jupyter/scipy-notebook:v0.4": rpc error: code = 2 desc = Error response from daemon: {"message":"manifest for jupyter/scipy-notebook:v0.4 not found"}
+      ...
+
+It seems there is indeed something wrong with the Docker image. Let's confirm
+this by getting another view on the events that have transpired in the pod.
+
+``kubectl logs``
+~~~~~~~~~~~~~~~~
+If you only want to see the latest logs for a pod, use the following command::
+
+    kubectl --namespace=jhub logs <POD_NAME>
+
+This will output a list of the recent events for the pod. Parse these logs
+to see if something is generating an error.
+
+**Real world scenario:** In our case, we get this line back::
+
+    $ kubectl --namespace=jhub logs jupyter-choldgraf
+    Error from server (BadRequest): container "notebook" in pod "jupyter-choldgraf" is waiting to start: trying and failing to pull image
+
+Now we are sure that something is wrong with our Docker file. Let's check
+our ``config.yaml`` file for the section where we specify the user's
+Docker image. Here we see our problem::
+
+  singleuser:
+  image:
+      name:
+          jupyter/scipy-notebook
+
+We haven't specified a ``tag`` for our Docker image! Not specifying a tag
+will cause it to default to ``v0.4``, which isn't what we want and is causing
+the pod to fail.
+
+To fix this, let's add a tag to our ``config.yaml`` file::
+
+  singleuser:
+  image:
+      name:
+          jupyter/scipy-notebook
+      tag:
+          ae885c0a6226
+
+Then run a helm upgrade::
+
+    helm upgrade jhub jupyterhub/jupyterhub --version=v0.4 -f config.yaml
+
+where ``jhub`` is the Kubernetes namespace (substitute the namespace that you
+chose during setup).
+
+.. note::
+
+   Depending on the size of the Docker image, this may take a while to complete.
+
+Right after you run this command, let's once again list the pods in our
+deployment::
+
+  $ kubectl --namespace=jhub get pod
+  NAME                                READY     STATUS              RESTARTS   AGE
+  hub-deployment-2653507799-r7wf8     0/1       ContainerCreating   0          31s
+  hub-deployment-3311438805-xnfvp     1/1       Terminating         0          14m
+  jupyter-choldgraf                   0/1       ImagePullBackOff    0          12m
+  proxy-deployment-1227971824-mn2wd   1/1       Running             0          5h
+
+Here we can see one ``hub-deployment`` pod being destroyed, and another (based
+on the upgraded helm chart) being created. We also see our broken user pod,
+which will only be upgraded if it is restarted. To do this, let's delete the

```

Comment:
"We also see our broken user pod, which will not be deleted automatically. Let's manually delete it so a newer working pod can be started"